docs/x509/reference.rst

```diff
@@ -871,11 +855,8 @@ X.509 CSR (Certificate Signing Request) Object
         The public key associated with the request.
 
         :returns: One of
-            :class:`~cryptography.hazmat.primitives.asymmetric.rsa.RSAPublicKey`,
-            :class:`~cryptography.hazmat.primitives.asymmetric.dsa.DSAPublicKey`,
-            :class:`~cryptography.hazmat.primitives.asymmetric.ec.EllipticCurvePublicKey`,
-            :class:`~cryptography.hazmat.primitives.asymmetric.ed25519.Ed25519PublicKey` or
-            :class:`~cryptography.hazmat.primitives.asymmetric.ed448.Ed448PublicKey`.
+            :data:`~cryptography.hazmat.primitives.asymmetric.types.CertificatePublicKeyTypes`.
+            # FIXME is this correct?
```

Comment:
The type in code of `CertificateSigningRequest.public_key()` uses `CertificatePublicKeyTypes`, but documentation instead matches `CertificateIssuerPublicKeyTypes`. Not sure which is correct.

Reply:
This can be `CertificatePublicKeyTypes` (it's allowable to have X25519/X448 in this).